gammapy/maps/axes.py

```diff
@@ -1797,41 +1818,45 @@ class TimeMapAxis:
         coordinates.  For now only 'lin' is supported.
     """
     node_type = "intervals"
+
     def __init__(self, edges_min, edges_max, reference_time, name="time", interp="lin"):
         self._name = name
 
         edges_min = u.Quantity(edges_min, ndmin=1)
         edges_max = u.Quantity(edges_max, ndmin=1)
 
-        reference_time = Time(reference_time)
-
-        invalid = [_.unit.name for _ in (edges_min, edges_max) if not _.unit.is_equivalent("d")]
-        if len(invalid)>0:
-            raise TypeError(f"TimeAxis edges must be time-like quantities. Got {invalid}")
+        if not edges_min.unit.is_equivalent("s"):
+            raise ValueError(f"Time edges min must have a valid time unit, got {edges_min.unit}")
 
-        # Note: flatten is there to deal with scalr Time objects
-        if not len(edges_min.flatten()) == len(edges_max.flatten()):
-            raise ValueError("Time min and time max must have the same length.")
+        if not edges_max.unit.is_equivalent("s"):
+            raise ValueError(f"Time edges max must have a valid time unit, got {edges_max.unit}")
 
-        if not (np.all(edges_min == np.sort(edges_min))
-                and np.all(edges_max == np.sort(edges_max))):
-            raise ValueError("TimeAxis: edges values must be sorted")
+        if not edges_min.shape == edges_max.shape:
+            raise ValueError("Time min and time max must have the same shape,"
+                             f" got {edges_min.shape} and {edges_max.shape}.")
 
-        self._edges_min = u.Quantity(edges_min)
-        self._edges_max = u.Quantity(edges_max)
-        self._reference_time = reference_time
-        self._pix_offset = -0.5
-        self._nbin = len(edges_min.flatten())
+        if not np.all(edges_min == np.sort(edges_min)):
+            raise ValueError("Time edges min values must be sorted")
 
-        if self.nbin>1:
-            delta = self._edges_min[1:] - self._edges_max[:-1]
-            if np.any(delta.to_value("s")<0):
-                raise ValueError("TimeMapAxis: time intervals must not overlap.")
+        if not np.all(edges_max == np.sort(edges_max)):
+            raise ValueError("Time edges max values must be sorted")
 
         if interp != "lin":
-            raise ValueError("TimeMapAxis: non-linear scaling scheme are not supported yet.")
+            raise NotImplementedError(f"Non-linear scaling scheme are not supported yet, got {interp}")
+
+        self._edges_min = edges_min
+        self._edges_max = edges_max
+        self._reference_time = Time(reference_time)
+        self._pix_offset = -0.5
         self._interp = interp
 
+        delta = edges_min[1:] - edges_max[:-1]
```

Comment:
this will break for 1 bin entry.

Reply:
I think it's fine:
```python3
from astropy import units as u
import numpy as np

a = u.Quantity(1, ndmin=1)
delta = a[1:] - a[:-1]
print(np.any(delta < 0))
```
`delta` will be an empty array that always evaluates to `False`. For a single bin the bins cannot be overlapping...